src/main/java/interfaces2/Knight.java

```diff
@@ -0,0 +1,21 @@
+package interfaces2;
+
+public class Knight {
+    private Quest quest; //pole klasy o typie interfejsu, do tego pola przypisana zostanie wartość
+                         // w trakcie tworzenia nowego obiektu Knight
+
+    public Knight(Quest quest) { //konstruktor, w którego argumencie znajduje się
+                                 // interfejs, który jest typem danych, tak samo jak int., itd
+        this.quest = quest;
+    }
+
+    public void doQuest() {
+        this.quest.process();  // to pole wywołuje metodę interfejsu -  process()
```

Comment:
The word "this" indicates that you call a method process(), on-field Quest from the current instance object ("Knight"). 
e.x if you have:

class Counter {
  private int value = 0;

  int incrementValue(int value) {
      value = value + 1;
      return value;
   }
   
   int getValue() {
      return value;
   }


}

How do you think, if you create a new Counter then call incrementValue(5) on that object, and what would be a value returned from getValue() :) -> try to do it :)

Reply:
let's discuss on the call